core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -164,6 +180,13 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
       }
 
       partitionLengths = writePartitionedData(mapOutputWriter);
+      if (checksumEnabled) {
+        long[] checksums = new long[numPartitions];
+        for (int i = 0; i < numPartitions; i ++) {
+          checksums[i] = partitionChecksums[i].getValue();
+        }
+        IndexShuffleBlockResolver.get().writeChecksumFile(shuffleId, mapId, checksums);
```

Comment:
How do we write the shuffle index file right now? In this method, we also calculate the partition lengths but we don't write the index file immediately.

Reply:
Oh actually we did, but it's done by `ShuffleMapOutputWriter.commitAllPartitions`. Does this checksum file work with custom shuffle extensions?